### PR TITLE
Implement coala-ci

### PR DESCRIPTION
**Pull request by yukiisbored**
Add linters for java source files, Ant build scripts, README files,
and the configuration files for linters.

### CODE REVIEW

**Comment by yukiisbored** [comment]
This Pull Request is being done as part of the Google Code-In task https://github.com/coala/coala/wiki/Google-Code-In-Task-Implement-coala .

The mentors are : @jayvdb , @sils
In addition to normal review comments, collaborators are welcome to come and discuss the task at https://gitter.im/coala/coala .

File: README.md [hunk 1]
```diff
@@ -1,31 +1,63 @@
...
-__That [processing-bugs](https://github.com/processing-bugs) fella is a damn liar.__ 
-The issues list has been imported from Google Code, so there are many spurious references 
-amongst them since the numbering changed. Basically, any time you see references to 
-changes made by [processing-bugs](https://github.com/processing-bugs), it may be somewhat suspect.
+# Processing
+
+This is the official source code for the [Processing][processing site] Development Environment (PDE)
+, the “core” and the libraries that are included with the [download][processing download].
+
+**I've found a bug!**
```
Comment (sils) [inline README.md]: @yukiisbored their standard seems to use `__` for bold text, not `**`, I'd recommend to configure markdownbear to their preferences. (Also applies for headings and probably more.)

File: .coafile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+[xml]
+bears = XMLBear
+files = app/build.xml, core/build.xml, java/build.xml, build/build.xml,
+        .checkstyle.xml
+
+[java]
+bears = CheckstyleBear
+files = core/src/**.java, app/src/**.java, java/src/**.java
+checkstyle_configs = .checkstyle.xml
```
Comment (sils) [inline .coafile]: it wasn't possible to do it with the coala given settings?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+language: generic
+services: docker
+script: >
+        docker run --volume=$(pwd):/app --workdir=/app coala/base:pre
+        coala -m NORMAL -C
```
Comment (sils) [inline .travis.yml]: maybe instead of m and C use the long forms for readability here

Comment (yukiisbored) [inline .travis.yml]: Okay, fixed it on 10972a3

File: .coafile (hunk 2 above)
Comment (sils) [inline .coafile]: You mean CheckstyleBear?

File: .coafile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+[default]
+bears = FilenameBear
+files = README.md, core/README.md, core/src/**.java, app/src/**.java,
+        java/src/**.java
+file_naming_convention = pascal
+
+[markdown]
+bears = MarkdownBear, LineLengthBear, SpaceConsistencyBear
+files = README.md, core/README.md
```
Comment (sils) [inline .coafile]: wouldn't it be simpler to use `**.md` and define the `ignore` properly in the `default` section?

Comment (yukiisbored) [inline .coafile]: Okay, fixed it on 15dee8e2

**Comment by jayvdb** [comment]
Please add a link to your travis build of this commit, so the maintainers can see what it will look like when green.

**Comment by yukiisbored** [comment]
Here's the link for the Travis-CI build for 15dee8e: https://travis-ci.org/yukiisbored/processing/builds/185449799

**Comment by benfry** [comment]
Some context here? What's this about?

**Comment by ishanSrt** [comment]
@benfry this is a configuration file for the project https://github.com/coala/coala which is a tool for lifting and fixing your code to detect inconsistencies, you can consider adding coala to your CI to detect inconsistencies in future commits. This config file is customised according to your CI and is done as a part of a Google Code In task https://github.com/coala/coala/wiki/Google-Code-In-Task-Implement-coala 

ping yuki if you are interested in this, so that he can update this to the current code style of your project if there were major changes since the date the PR was made.

**Comment by jeremydouglass** [comment]
@benfry last summer we used coala for Processing.R for code linting, and as checkers go it was a fairly pleasant experience. 

Here was the setup Ce Gao developed (points to XML file for Java Checkbear):

https://github.com/processing-r/Processing.R/blob/f3a0927367ea74fb40ec71f0c102f026eeafb6a9/.coafile

**Comment by yukiisbored** [comment]
Hi @benfry, This PR was a part of a [Google Code-in](https://codein.withgoogle.com) 2016 task for the coala organization (when it was under the FOSSASIA umbrella).

Basically, [coala](https://coala.io) is a common interface to utilize various linting utilities and making it easy to configure and use a bunch of linters.

If you want it to be part of processing now, Feel free to message me so I can allocate some time in the future to update it.

**Comment by benfry** [comment]
Resolved with Travis CI via https://github.com/processing/processing4/pull/1 and continued development happening at https://github.com/processing/processing4

**Pull request closed without merging**